pkg/operation/botanist/component/extensions/worker/worker.go

```diff
@@ -293,6 +300,22 @@ func (w *worker) Wait(ctx context.Context) error {
 		w.waitInterval,
 		w.waitSevereThreshold,
 		w.waitTimeout,
+		nil,
+	)
+}
+
+// WaitUntilWorkerStatusMachineDeploymentsUpdate waits until the worker status is updated with the latest machineDeployment slice.
+func (w *worker) WaitUntilWorkerStatusMachineDeploymentsUpdate(ctx context.Context) error {
```

Comment:
Can you add unit tests for this function?

Reply:
addressed in [e26191c](https://github.com/gardener/gardener/pull/7774/commits/e26191c2e5ac76a299562a991fe601d380810a5d)